### PR TITLE
FEAT: Matrix invert function

### PR DESCRIPTION
**Pull request by cemoody**
Implemented a batch matrix inversion function. Addresses part of #575, but that still needs to support a determinant function

### CODE REVIEW

**Comment by cemoody** [comment]
I'm not sure why the checks are failing -- the command `nosetests -a '!gpu' tests/chainer_tests` works on my machine.

File: chainer/functions/math/inv.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
+import six
+import numpy.linalg
+
+import cupy
```
Comment (okuta) [inline chainer/functions/math/inv.py]: Please remove this line. Chainer must not directly import the cupy. Use the `chainer.cuda.cupy`.
For example,
https://github.com/pfnet/chainer/blob/83800a77b9118fa4d7e98b03056a95f488e96a51/chainer/functions/array/select_item.py#L49

File: tests/chainer_tests/functions_tests/math_tests/test_unary.py [hunk 2]
```diff
@@ -0,0 +1,171 @@
+import unittest
+
+import numpy
+
+import chainer
+import cupy
```
Comment (okuta) [inline tests/chainer_tests/functions_tests/math_tests/test_unary.py]: Remove this line.

**Comment by cemoody** [comment]
OK, thanks for reviewing @okuta. I fixed the failing tests and changed `cupy` -> `chainer.cuda.cupy`.

**Comment by okuta** [comment]
@cemoody Do you have time to work on this PR? Since it requires some knowledge to resolve conflicts, we will help you to fix the code.

I am sorry to make you hurry, but we will work on sending you PR if no response is returned until the end of this week (1/8).

**Comment by cemoody** [comment]
Yes, please do send a PR. As you recognized, it's difficult for me to resolve the conflicts. 

If you send a PR, I'll be happy to review! Many thanks :)

**Comment by okuta** [comment]
I'm sorry it takes a long time to fix.
Please merge the following PR.
https://github.com/cemoody/chainer/pull/2

**Comment by okuta** [comment]
LGTM!

**Pull request merged**